File: VRTInterstitialCustomEventTapjoy.m
//Header
#import "VRTInterstitialCustomEventTapjoy.h"

//Dependencies
@import Tapjoy;

@interface VRTInterstitialCustomEventTapjoy() <TJPlacementDelegate, TJPlacementVideoDelegate>
@property TJPlacement *tjPlacement;
@end

@implementation VRTInterstitialCustomEventTapjoy

- (void) loadInterstitialAd {
    NSString *placementName = [self.customEventConfig.thirdPartyCustomEventData objectForKey:@"adUnitId"];
    
    if (placementName == nil) {
        VRTError *error = [VRTError errorWithCode:VRTErrorCodeCustomEvent message:@"No placementName"];
        [self.customEventLoadDelegate customEventFailedToLoadWithError:error];
        return;
    }
    
    self.tjPlacement = [[TJPlacement alloc] init];
    self.tjPlacement.placementName = placementName;
    self.tjPlacement.delegate = self;
    self.tjPlacement.videoDelegate = self;
    [self.tjPlacement requestContent];
}

- (void) showInterstitialAd {
    UIViewController *vc = [self.viewControllerDelegate vrtViewControllerForModalPresentation];
    [self.tjPlacement showContentWithViewController:vc];
}


#pragma mark - TJPlacementDelegate
- (void)requestDidSucceed:(TJPlacement *)placement {
    //No VRT Analog
}

- (void)requestDidFail:(TJPlacement *)placement error:(nullable NSError *)error {
    [self.customEventLoadDelegate customEventFailedToLoadWithError:error];
}

- (void)contentIsReady:(TJPlacement *)placement {
    [self.customEventLoadDelegate customEventLoaded];
}

- (void)contentDidAppear:(TJPlacement *)placement {
    [self.customEventShowDelegate customEventDidPresentModal:VRTModalTypeInterstitial];
}

- (void)contentDidDisappear:(TJPlacement *)placement {
    [self.customEventShowDelegate customEventDidDismissModal:VRTModalTypeInterstitial];
}

- (void)didClick:(TJPlacement *)placement {
    [self.customEventShowDelegate customEventClicked];
}

- (void)placement:(TJPlacement *)placement
didRequestPurchase:(nullable TJActionRequest *)request
        productId:(nullable NSString *)productId {
    //No VRT Analog
}

- (void)placement:(TJPlacement *)placement
 didRequestReward:(nullable TJActionRequest *)request
           itemId:(nullable NSString *)itemId
         quantity:(int)quantity {
    //No VRT Analog
}

#pragma mark - TJPlacementVideoDelegate
- (void)videoDidStart:(TJPlacement *)placement {
    [self.customEventShowDelegate customEventVideoStarted];
}

- (void)videoDidComplete:(TJPlacement *)placement {
    [self.customEventShowDelegate customEventVideoCompleted];
}

- (void)videoDidFail:(TJPlacement *)placement error:(nullable NSString *)errorMsg {
    VRTError *vrtError = [VRTError errorWithCode:VRTErrorCodeCustomEvent message:errorMsg];
    [self.customEventLoadDelegate customEventFailedToLoadWithError:vrtError];
}


@end

File: VRTInterstitialCustomEventTapjoy.swift
//  Converted to Swift 5.8.1 by Swiftify v5.8.26605 - https://swiftify.com/
//Header
import Tapjoy

//Tapjoy Banner Adapter, Vrtcal as Primary

class VRTInterstitialCustomEventTapjoy: VRTAbstractInterstitialCustomEvent, TJPlacementDelegate, TJPlacementVideoDelegate {
    private var tjPlacement: TJPlacement?

    func loadInterstitialAd() {
        let placementName = customEventConfig.thirdPartyCustomEventData["adUnitId"] as? String

        if placementName == nil {
            let error = VRTError(code: VRTErrorCodeCustomEvent, message: "No placementName")
            customEventLoadDelegate.customEventFailedToLoadWithError(error)
            return
        }

        tjPlacement = TJPlacement()
        tjPlacement?.placementName = placementName
        tjPlacement?.delegate = self
        tjPlacement?.videoDelegate = self
        tjPlacement?.requestContent()
    }

    func showInterstitialAd() {
        let vc = viewControllerDelegate.vrtViewControllerForModalPresentation()
        tjPlacement?.showContent(with: vc)
    }

    // MARK: - TJPlacementDelegate

    func requestDidSucceed(_ placement: TJPlacement?) {
        //No VRT Analog
    }

    func requestDidFail(_ placement: TJPlacement?, error: Error?) {
        customEventLoadDelegate.customEventFailedToLoadWithError(error)
    }

    func contentIsReady(_ placement: TJPlacement?) {
        customEventLoadDelegate.customEventLoaded()
    }

    func contentDidAppear(_ placement: TJPlacement?) {
        customEventShowDelegate.customEventDidPresentModal(VRTModalTypeInterstitial)
    }

    func contentDidDisappear(_ placement: TJPlacement?) {
        customEventShowDelegate.customEventDidDismissModal(VRTModalTypeInterstitial)
    }

    func didClick(_ placement: TJPlacement?) {
        customEventShowDelegate.customEventClicked()
    }

    func placement(
        _ placement: TJPlacement?,
        didRequestPurchase request: TJActionRequest?,
        productId: String?
    ) {
        //No VRT Analog
    }

    func placement(
        _ placement: TJPlacement?,
        didRequestReward request: TJActionRequest?,
        itemId: String?,
        quantity: Int
    ) {
        //No VRT Analog
    }

    // MARK: - TJPlacementVideoDelegate

    func videoDidStart(_ placement: TJPlacement?) {
        customEventShowDelegate.customEventVideoStarted()
    }

    func videoDidComplete(_ placement: TJPlacement?) {
        customEventShowDelegate.customEventVideoCompleted()
    }

    func videoDidFail(_ placement: TJPlacement?, error errorMsg: String?) {
        let vrtError = VRTError(code: VRTErrorCodeCustomEvent, message: errorMsg)
        customEventLoadDelegate.customEventFailedToLoadWithError(vrtError)
    }
}

//Dependencies